plat/Config.uk

```diff
@@ -115,3 +115,12 @@ config FPSIMD
 	depends on ARCH_ARM_64
 	help
 		Enable support FPU usage in application
+
+# FIXME: Temporary workaround to display a warning when more than one platform
```

Comment:
Yes, the idea is to change the menu in order to let only one platform selectable. Because we also support platforms from external repos, we need to sit down first and come up with a solution on how we can properly auto-generate the `choice`. Because we originally supported multiple platform images at the same time, we could simply include each platform's `Config.uk` and compile the menu that way. Problem is that with a `choice` this is a bit more difficult: we get basically get two separated blocks in KConfig: 1) the `choice` listing all platforms, 2) the configuration items of each platform and they must be encapsulated with `if` condition so that they are only visible depending on the chosen platform. For building this mechanics, we need to be able to somehow extract the name of a platform form a platform repo.
Because all of this, the patch here is really for bridging the gap until we make it clean. Since we began already re-architecting the driver organization (wip, ongoing), we broke already the property that allowed us to compile multiple images at the same time. So this must be avoided that this can be configured and we won't go back to support multiple builds.

Reply:
> Maybe it's better to set depend PLAT_XEN from !PLAT_KVM and vice versa?

Hum, this is also an option how we could do this temporarily. However, transition a config from KVM to XEN and vise versa is less obvious with that. I guess this is mostly preference. What do you think? What do you think @michpappas ?

Reply:
Good arguments and I like your suggestion @michpappas. I will do the following: I am implementing for this PR @oleksiimoisieiev's suggestion with `PLAT_XEN depends on !PLAT_KVM` (and vice versa) to cover the configuration problem we have with the current tree.
The build system does stop compiling if we have too many platforms configured (e.g., when external platforms are added and selected): so this is a okay~ish behavior: at least nothing invalid is compiled. The last missing part with the menu, I would do in a separate PR because of the amount of changes that this involves is not covered by the intention of this PR anymore. I think there is also some discussion still open to finally conclude.

Reply:
Update: This change causes a KConfig circular dependency 😭 :
```diff
diff --git a/plat/kvm/Config.uk b/plat/kvm/Config.uk
index c6640cef3..8ec06d16f 100644
--- a/plat/kvm/Config.uk
+++ b/plat/kvm/Config.uk
@@ -1,6 +1,6 @@
 menuconfig PLAT_KVM
        bool "KVM guest"
-       default n
+       depends on !PLAT_XEN
        depends on (ARCH_X86_64 || ARCH_ARM_64)
        select LIBUKDEBUG
        select LIBUKBOOT
diff --git a/plat/xen/Config.uk b/plat/xen/Config.uk
index 81dbe2f2f..cfdcd34a6 100644
--- a/plat/xen/Config.uk
+++ b/plat/xen/Config.uk
@@ -1,6 +1,6 @@
 menuconfig PLAT_XEN
        bool "Xen guest image"
-       default n
+       depends on !PLAT_KVM
        depends on ((ARCH_X86_32 && !HAVE_SMP) || (ARCH_X86_64 && !HAVE_SMP) || (ARCH_ARM_32 && !HAVE_SMP) || (ARCH_ARM_64 && !HAVE_SMP))
        depends on ((ARCH_X86_32 && !HAVE_SYSCALL) || (ARCH_X86_64 && !HAVE_SYSCALL) || (ARCH_ARM_32 && !HAVE_SYSCALL) || (ARCH_ARM_64 && !HAVE_SYSCALL))
        depends on !HAVE_PAGING
```

Hum, I would say that we keep then this warning message for now and solve it with a separate PR where we clean-up the platform configuration menu. It looks like a `choice` is needed.

Reply:
> The last missing part with the menu, I would do in a separate PR because of the amount of changes that this involves is not covered by the intention of this PR anymore

Ok from my side 👍🏼